### PR TITLE
Switch to axum 0.6

### PR DESCRIPTION
**Pull request by FelixWeis**
A generic answer as to why we're switching can be found here:

[phind.com/search?q=axum+vs+actix-web+in+early+2023](https://www.phind.com/search?cache=01eaec65-7e11-4bd4-a24e-7ed56042744c)

For this project specifically, to deliver all the interactive elements we have in mind, we are significantly depending on asynchronous functionality. Although utilizing two primary frameworks with < 1.0 semver might present some challenges.

Nevertheless, both Leptos and Axum are advancing rapidly. The long-term benefits for developer ergonomics, particularly with the evolving tracing capabilities of tokio, appear promising.

### tasks
- [x] base 
- [x] `tokio::console`
- [x] `axum::extract::State`
- [x] process prometheus
- [x] per request prometheus

### testing this changeset
- everything should work as before
- using `RUST_LOG=trace` enables [`tokio-console`](https://github.com/tokio-rs/console), to examining all async tasks performance
- prometheus endpoints. 
  - `127.0.0.1:3000/metrics` per request histogram
  - `127.0.0.1:9100/metrics` process mem/cpu usage

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,144 +1,208 @@
...
-            dbg!(&backend_bitcoin_core);
-            if std::env::var("RUST_BACKTRACE").is_ok() {
-                let mut buffer = String::new();
-                let stdin = std::io::stdin(); // We get `Stdin` here.
-                println!("Press enter to start with config above. Crtl+C to abort.");
-                stdin.read_line(&mut buffer)?;
+// #[get("/style.css")]
+// async fn css() -> impl Responder {
+//     actix_files::NamedFile::open_async("./style/output.css").await
+// }
```
Comment (sectore) [inline src/main.rs:66]: Yep, that's an Actix artifact and can be removed - `css` is already linked in `App`.

File: src/fallback.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        handler(req).await.into_response()
+    }
+}
+#[cfg(feature = "ssr")]
+async fn get_static_file(uri: Uri, root: &str) -> Result<Response<BoxBody>, (StatusCode, String)> {
+    let req = Request::builder()
+        .uri(uri.clone())
+        .body(Body::empty())
+        .unwrap();
+    let root_path = format!("{root}");
```
Comment (sectore) [inline src/fallback.rs:44]: Q. Why is this needed?

File: src/main.rs [hunk 3]
```diff
@@ -1,144 +1,208 @@
...
+    //         .service(Files::new("/punks/", "/tmp/punks"))
+    //         .service(Files::new("/", &site_root))
+    //         //.wrap(middleware::Compress::default())
+    // })
+    // .bind(&addr)?
+    // .run()
+    // .await
+
+    // let addr = "[::]:3000".parse().unwrap();
+    let addr = "[::]:3000".parse::<std::net::SocketAddr>().unwrap();
```
[the remark this answers is not in the document]
Comment (FelixWeis) [reply src/main.rs]: good call

File: src/fallback.rs (hunk 2 above)
Comment (FelixWeis) [reply src/fallback.rs:44]: not sure, I just copied from axum example

**Review by fjahr — APPROVED** [review]
tACK

I only skimmed the code but tested with regtest and mainnet and didn't see any issues.
